MVC/EShopDemo/Controllers/CategoriesController.cs

```diff
@@ -0,0 +1,139 @@
+﻿using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+using EShopDemo.Models;
+
+namespace EShopDemo.Controllers
+{
+    public class CategoriesController : Controller
+    {
+        private readonly EShopContext _context;
+
+        public CategoriesController(EShopContext context)
+        {
+            _context = context;
+        }
+
+        public async Task<IActionResult> Index()
+        {
+            return View(await _context.Categories.ToListAsync());
+        }
+
+        public async Task<IActionResult> Details(int? id)
+        {
+            if (id == null)
+            {
+                return NotFound();
+            }
+
+            var category = await _context.Categories
+                .FirstOrDefaultAsync(m => m.CategoryId == id);
+            if (category == null)
+            {
+                return NotFound();
+            }
+
+            return View(category);
+        }
+
+        public IActionResult Create()
+        {
+            return View();
+        }
+
+        [HttpPost]
+        [ValidateAntiForgeryToken]
```

Comment:
Not sure this is required since I don't see any anti forgery request validation. Should it be added?